autogpt_platform/backend/backend/data/credit.py

```diff
@@ -431,15 +609,20 @@ async def _top_up_credits(
                     },
                 )
                 if payment_intent.status == "succeeded":
-                    await self._enable_transaction(
-                        transaction_key=transaction_key,
-                        user_id=user_id,
-                        metadata=Json({"payment_intent": payment_intent}),
-                    )
-                    return
-
-        raise ValueError(
-            f"Out of {len(payment_methods)} payment methods tried, none is supported"
+                    successful_transaction = Json({"payment_intent": payment_intent})
+                    new_transaction_key = payment_intent.id
+                    break
+
+        if not successful_transaction:
+            raise ValueError(
+                f"Out of {len(payment_methods)} payment methods tried, none is supported"
+            )
+
+        await self._enable_transaction(
+            transaction_key=transaction_key,
+            new_transaction_key=new_transaction_key,
+            user_id=user_id,
+            metadata=successful_transaction,
```

Comment:
The change on this function is also reliability enhancement, making sure we are first creating an inactive top-up request, then call Stripe, then enable it if it succeed.